Lib/dataclasses.py

```diff
@@ -1235,13 +1230,38 @@ def _update_func_cell_for__class__(f, oldcls, newcls):
     # Fix the cell to point to the new class, if it's already pointing
     # at the old class.  I'm not convinced that the "is oldcls" test
     # is needed, but other than performance can't hurt.
-    closure = f.__closure__[idx]
-    if closure.cell_contents is oldcls:
-        closure.cell_contents = newcls
+    cell = f.__closure__[idx]
+    if cell.cell_contents is oldcls:
+        cell.cell_contents = newcls
         return True
     return False
 
 
+def _find_inner_functions(obj, _seen=None, _depth=0):
+    if _seen is None:
+        _seen = set()
+    if id(obj) in _seen:
+        return None
+    _seen.add(id(obj))
+
+    _depth += 1
+    # Normally just inspection of a descriptor object itself should be enough,
+    # and we should encounter the function as its attribute,
+    # but in case function was wrapped (e.g. functools.partition was used),
+    # we want to dive at least one level deeper.
+    if _depth > 2:
+        return None
+
+    for attr in dir(obj):
+        value = getattr(obj, attr, None)
```

Comment:
This case is always complex, since `getattr` can evaluate properties and other objects. There are two ways of hanling this case:
1. Allow all exceptions to raise
2. Hide all exceptions from user here

I am not quite sure which one is better here.

Reply:
Thanks for bringing this to my attention! 

After some hacking, I've decided to go with the third option: removing any possibility of user defined code execution.